Build failure on openSUSE Tumbleweed (gcc 9.2.1)

Build of v0.5.1-beta/master fails to build on openSUSE Tumbleweed systems...
```
...
 52%] Building CXX object src/CMakeFiles/rigel_core.dir/game_logic/entity_factory.cpp.o
In file included from /home/mhauke/tmp/RigelEngine/src/game_logic/entity_factory.cpp:413:
/home/mhauke/tmp/RigelEngine/src/game_logic/entity_configuration.ipp: In member function ‘void rigel::game_logic::EntityFactory::configureEntity(entityx::Entity, rigel::data::ActorID, const BoundingBox&)’:
/home/mhauke/tmp/RigelEngine/src/game_logic/entity_configuration.ipp:1749:72: error: use of deleted function ‘std::variant<_Types>::variant() [with _Types = {rigel::game_logic::behaviors::SpikedGreenCreature::Awakening, rigel::game_logic::behaviors::SpikedGreenCreature::Waiting, rigel::game_logic::behaviors::SpikedGreenCreature::Pouncing, rigel::game_logic::behaviors::SpikedGreenCreature::Landing}]’
 1749 |       entity.assign<BehaviorController>(behaviors::SpikedGreenCreature{});
      |                                                                        ^
In file included from /home/mhauke/tmp/RigelEngine/src/game_logic/effect_components.hpp:33,
                 from /home/mhauke/tmp/RigelEngine/src/game_logic/entity_factory.cpp:32:
/usr/include/c++/9/variant:1287:7: note: ‘std::variant<_Types>::variant() [with _Types = {rigel::game_logic::behaviors::SpikedGreenCreature::Awakening, rigel::game_logic::behaviors::SpikedGreenCreature::Waiting, rigel::game_logic::behaviors::SpikedGreenCreature::Pouncing, rigel::game_logic::behaviors::SpikedGreenCreature::Landing}]’ is implicitly deleted because the default definition would be ill-formed:
 1287 |       variant() = default;
      |       ^~~~~~~
/usr/include/c++/9/variant:1287:7: error: use of deleted function ‘constexpr std::_Enable_default_constructor<false, _Tag>::_Enable_default_constructor() [with _Tag = std::variant<rigel::game_logic::behaviors::SpikedGreenCreature::Awakening, rigel::game_logic::behaviors::SpikedGreenCreature::Waiting, rigel::game_logic::behaviors::SpikedGreenCreature::Pouncing, rigel::game_logic::behaviors::SpikedGreenCreature::Landing>]’
In file included from /usr/include/c++/9/optional:43,
                 from /usr/include/c++/9/bits/node_handle.h:39,
                 from /usr/include/c++/9/bits/hashtable.h:37,
                 from /usr/include/c++/9/unordered_map:46,
                 from /home/mhauke/tmp/RigelEngine/3rd_party/entityx/entityx/Event.h:17,
                 from /home/mhauke/tmp/RigelEngine/3rd_party/entityx/entityx/entityx.h:4,
                 from /home/mhauke/tmp/RigelEngine/src/engine/timing.hpp:22,
                 from /home/mhauke/tmp/RigelEngine/src/engine/visual_components.hpp:23,
                 from /home/mhauke/tmp/RigelEngine/src/game_logic/entity_factory.hpp:22,
                 from /home/mhauke/tmp/RigelEngine/src/game_logic/entity_factory.cpp:17:
/usr/include/c++/9/bits/enable_special_members.h:110:15: note: declared here
  110 |     constexpr _Enable_default_constructor() noexcept = delete;
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/mhauke/tmp/RigelEngine/src/game_logic/entity_factory.cpp:413:
/home/mhauke/tmp/RigelEngine/src/game_logic/entity_configuration.ipp:1822:65: error: use of deleted function ‘std::variant<_Types>::variant() [with _Types = {rigel::game_logic::behaviors::BossEpisode2::WarmingUp, rigel::game_logic::behaviors::BossEpisode2::FlyingRight, rigel::game_logic::behaviors::BossEpisode2::FlyingLeft, rigel::game_logic::behaviors::BossEpisode2::FallingDown, rigel::game_logic::behaviors::BossEpisode2::JumpingRight, rigel::game_logic::behaviors::BossEpisode2::JumpingLeft, rigel::game_logic::behaviors::BossEpisode2::RisingUp}]’
 1822 |       entity.assign<BehaviorController>(behaviors::BossEpisode2{});
      |                                                                 ^
In file included from /home/mhauke/tmp/RigelEngine/src/game_logic/effect_components.hpp:33,
                 from /home/mhauke/tmp/RigelEngine/src/game_logic/entity_factory.cpp:32:
/usr/include/c++/9/variant:1287:7: note: ‘std::variant<_Types>::variant() [with _Types = {rigel::game_logic::behaviors::BossEpisode2::WarmingUp, rigel::game_logic::behaviors::BossEpisode2::FlyingRight, rigel::game_logic::behaviors::BossEpisode2::FlyingLeft, rigel::game_logic::behaviors::BossEpisode2::FallingDown, rigel::game_logic::behaviors::BossEpisode2::JumpingRight, rigel::game_logic::behaviors::BossEpisode2::JumpingLeft, rigel::game_logic::behaviors::BossEpisode2::RisingUp}]’ is implicitly deleted because the default definition would be ill-formed:
 1287 |       variant() = default;
      |       ^~~~~~~
/usr/include/c++/9/variant:1287:7: error: use of deleted function ‘constexpr std::_Enable_default_constructor<false, _Tag>::_Enable_default_constructor() [with _Tag = std::variant<rigel::game_logic::behaviors::BossEpisode2::WarmingUp, rigel::game_logic::behaviors::BossEpisode2::FlyingRight, rigel::game_logic::behaviors::BossEpisode2::FlyingLeft, rigel::game_logic::behaviors::BossEpisode2::FallingDown, rigel::game_logic::behaviors::BossEpisode2::JumpingRight, rigel::game_logic::behaviors::BossEpisode2::JumpingLeft, rigel::game_logic::behaviors::BossEpisode2::RisingUp>]’
In file included from /usr/include/c++/9/optional:43,
                 from /usr/include/c++/9/bits/node_handle.h:39,
                 from /usr/include/c++/9/bits/hashtable.h:37,
                 from /usr/include/c++/9/unordered_map:46,
                 from /home/mhauke/tmp/RigelEngine/3rd_party/entityx/entityx/Event.h:17,
                 from /home/mhauke/tmp/RigelEngine/3rd_party/entityx/entityx/entityx.h:4,
                 from /home/mhauke/tmp/RigelEngine/src/engine/timing.hpp:22,
                 from /home/mhauke/tmp/RigelEngine/src/engine/visual_components.hpp:23,
                 from /home/mhauke/tmp/RigelEngine/src/game_logic/entity_factory.hpp:22,
                 from /home/mhauke/tmp/RigelEngine/src/game_logic/entity_factory.cpp:17:
/usr/include/c++/9/bits/enable_special_members.h:110:15: note: declared here
  110 |     constexpr _Enable_default_constructor() noexcept = delete;
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/mhauke/tmp/RigelEngine/src/game_logic/entity_factory.cpp:413:
/home/mhauke/tmp/RigelEngine/src/game_logic/entity_configuration.ipp:1891:64: error: use of deleted function ‘std::variant<_Types>::variant() [with _Types = {rigel::game_logic::behaviors::UnicycleBot::Waiting, rigel::game_logic::behaviors::UnicycleBot::Accelerating, rigel::game_logic::behaviors::UnicycleBot::Moving}]’
 1891 |       entity.assign<BehaviorController>(behaviors::UnicycleBot{});
      |                                                                ^
In file included from /home/mhauke/tmp/RigelEngine/src/game_logic/effect_components.hpp:33,
                 from /home/mhauke/tmp/RigelEngine/src/game_logic/entity_factory.cpp:32:
/usr/include/c++/9/variant:1287:7: note: ‘std::variant<_Types>::variant() [with _Types = {rigel::game_logic::behaviors::UnicycleBot::Waiting, rigel::game_logic::behaviors::UnicycleBot::Accelerating, rigel::game_logic::behaviors::UnicycleBot::Moving}]’ is implicitly deleted because the default definition would be ill-formed:
 1287 |       variant() = default;
      |       ^~~~~~~
/usr/include/c++/9/variant:1287:7: error: use of deleted function ‘constexpr std::_Enable_default_constructor<false, _Tag>::_Enable_default_constructor() [with _Tag = std::variant<rigel::game_logic::behaviors::UnicycleBot::Waiting, rigel::game_logic::behaviors::UnicycleBot::Accelerating, rigel::game_logic::behaviors::UnicycleBot::Moving>]’
In file included from /usr/include/c++/9/optional:43,
                 from /usr/include/c++/9/bits/node_handle.h:39,
                 from /usr/include/c++/9/bits/hashtable.h:37,
                 from /usr/include/c++/9/unordered_map:46,
                 from /home/mhauke/tmp/RigelEngine/3rd_party/entityx/entityx/Event.h:17,
                 from /home/mhauke/tmp/RigelEngine/3rd_party/entityx/entityx/entityx.h:4,
                 from /home/mhauke/tmp/RigelEngine/src/engine/timing.hpp:22,
                 from /home/mhauke/tmp/RigelEngine/src/engine/visual_components.hpp:23,
                 from /home/mhauke/tmp/RigelEngine/src/game_logic/entity_factory.hpp:22,
                 from /home/mhauke/tmp/RigelEngine/src/game_logic/entity_factory.cpp:17:
/usr/include/c++/9/bits/enable_special_members.h:110:15: note: declared here
  110 |     constexpr _Enable_default_constructor() noexcept = delete;
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/mhauke/tmp/RigelEngine/src/game_logic/entity_factory.cpp:413:
/home/mhauke/tmp/RigelEngine/src/game_logic/entity_configuration.ipp:2170:65: error: use of deleted function ‘std::variant<_Types>::variant() [with _Types = {rigel::game_logic::behaviors::LavaFountain::Waiting, rigel::game_logic::behaviors::LavaFountain::Erupting}]’
 2170 |       entity.assign<BehaviorController>(behaviors::LavaFountain{});
      |                                                                 ^
In file included from /home/mhauke/tmp/RigelEngine/src/game_logic/effect_components.hpp:33,
                 from /home/mhauke/tmp/RigelEngine/src/game_logic/entity_factory.cpp:32:
/usr/include/c++/9/variant:1287:7: note: ‘std::variant<_Types>::variant() [with _Types = {rigel::game_logic::behaviors::LavaFountain::Waiting, rigel::game_logic::behaviors::LavaFountain::Erupting}]’ is implicitly deleted because the default definition would be ill-formed:
 1287 |       variant() = default;
      |       ^~~~~~~
/usr/include/c++/9/variant:1287:7: error: use of deleted function ‘constexpr std::_Enable_default_constructor<false, _Tag>::_Enable_default_constructor() [with _Tag = std::variant<rigel::game_logic::behaviors::LavaFountain::Waiting, rigel::game_logic::behaviors::LavaFountain::Erupting>]’
In file included from /usr/include/c++/9/optional:43,
                 from /usr/include/c++/9/bits/node_handle.h:39,
                 from /usr/include/c++/9/bits/hashtable.h:37,
                 from /usr/include/c++/9/unordered_map:46,
                 from /home/mhauke/tmp/RigelEngine/3rd_party/entityx/entityx/Event.h:17,
                 from /home/mhauke/tmp/RigelEngine/3rd_party/entityx/entityx/entityx.h:4,
                 from /home/mhauke/tmp/RigelEngine/src/engine/timing.hpp:22,
                 from /home/mhauke/tmp/RigelEngine/src/engine/visual_components.hpp:23,
                 from /home/mhauke/tmp/RigelEngine/src/game_logic/entity_factory.hpp:22,
                 from /home/mhauke/tmp/RigelEngine/src/game_logic/entity_factory.cpp:17:
/usr/include/c++/9/bits/enable_special_members.h:110:15: note: declared here
  110 |     constexpr _Enable_default_constructor() noexcept = delete;
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [src/CMakeFiles/rigel_core.dir/build.make:869: src/CMakeFiles/rigel_core.dir/game_logic/entity_factory.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:446: src/CMakeFiles/rigel_core.dir/all] Error 2
make: *** [Makefile:141: all] Error 2

```
gcc version 9.2.1 20200128 [revision 83f65674e78d97d27537361de1a9d74067ff228d] (SUSE Linux) 

Full buildlog:
[RigelEngine-openSUSE-Tumbleweed-broken-build.txt](https://github.com/lethal-guitar/RigelEngine/files/4208984/RigelEngine-openSUSE-Tumbleweed-broken-build.txt)

--- a/src/game_logic/enemies/boss_episode_2.hpp
+++ b/src/game_logic/enemies/boss_episode_2.hpp
@@ -91,7 +91,7 @@ struct BossEpisode2 {
     const base::Point<float>& inflictorVelocity,
     entityx::Entity entity);
 
-  State mState;
+  State mState = WarmingUp{};
   int mStartingHeight = 0;
   int mCoolDownFrames = 0;
   bool mDestructionPending = false;

--- a/src/game_logic/enemies/spiked_green_creature.hpp
+++ b/src/game_logic/enemies/spiked_green_creature.hpp
@@ -74,7 +74,7 @@ struct SpikedGreenCreature {
     const GlobalDependencies& d,
     entityx::Entity entity);
 
-  State mState;
+  State mState = Awakening{};
 };
 
 }

--- a/src/game_logic/enemies/unicycle_bot.hpp
+++ b/src/game_logic/enemies/unicycle_bot.hpp
@@ -52,7 +52,7 @@ struct UnicycleBot {
     bool isOnScreen,
     entityx::Entity entity);
 
-  State mState;
+  State mState = Waiting{};
   int mFramesUntilNextTurn = 0;
 };
 

--- a/src/game_logic/hazards/lava_fountain.cpp
+++ b/src/game_logic/hazards/lava_fountain.cpp
@@ -93,7 +93,7 @@ void LavaFountain::update(
       if (state.mSequenceIndex == 0) {
         entity.assign<game_logic::components::PlayerDamaging>(1);
       }
-      if (state.mSequenceIndex == ERUPTION_SEQUENCE.size()) {
+      if (state.mSequenceIndex == static_cast<int>(ERUPTION_SEQUENCE.size())) {
         resetBbox();
         entity.remove<game_logic::components::PlayerDamaging>();
 

--- a/src/game_logic/hazards/lava_fountain.hpp
+++ b/src/game_logic/hazards/lava_fountain.hpp
@@ -58,7 +58,7 @@ struct LavaFountain {
     const engine::components::Sprite& sprite,
     const base::Vector& positionInScreenSpace);
 
-  State mState;
+  State mState = Waiting{};
 };
 
 }

